### PR TITLE
Bug/cleaning up little issues

### PR DESCRIPTION
**Pull request by reisr3**
- Added contact information to contract form
- Fixed auth with Firebase

### CODE REVIEW

File: frontend/src/utils/schemas/v1.ts [hunk 1]
```diff
@@ -12,6 +12,7 @@ export const schema_v1: JSONSchemaType<IIPFSJobMetaData> = {
     version: { type: 'integer', minimum: 1 },
     title: { type: 'string', minLength: 3, maxLength: 120 },
     description: { type: 'string', minLength: 3, maxLength: 24576 },
+    contactInformation: { type: 'string', minLength: 1, maxLength: 2457 },
```
Comment (deweller) [inline frontend/src/utils/schemas/v1.ts:15]: Should this be a maxLength of 250 to match yup?

File: frontend/src/components/wallet/useWalletProvider.ts [hunk 2]
```diff
@@ -81,7 +81,7 @@ export const useWalletProvider = () => {
           method: 'eth_accounts',
         });
 
-        if (result.length > 0) {
+        if (result?.length > 0) {
```
Comment (deweller) [inline frontend/src/components/wallet/useWalletProvider.ts:84]: Metamask will always return an array here.

Same question as above.  Is this necessary?

File: frontend/src/pages/MyContracts.tsx [hunk 3]
```diff
@@ -14,7 +14,7 @@ export const MyContracts = () => {
       <SortFilterHeading heading="My Contracts" />
       {isLoading ? (
         <Loader />
-      ) : jobs.length > 0 ? (
+      ) : jobs?.length > 0 ? (
```
Comment (deweller) [inline frontend/src/pages/MyContracts.tsx:17]: jobs will be `IJobData[]`.

Same question as above.  Is this necessary?

File: frontend/src/components/smart-contracts/hooks/useERC20Events.ts [hunk 4]
```diff
@@ -45,7 +45,7 @@ export const lookupERC20Approval = async (
   erc20Contract: ERC20
 ) => {
   // query the blockchain contract to see if we have approval
-  if (account.length) {
+  if (account?.length) {
```
[the remark this answers is not in the document]
Comment (reisr3) [reply frontend/src/components/smart-contracts/hooks/useERC20Events.ts:48]: No not really :D mostly superstition. I got a `cannot read property of length` error in `dev` and couldn't see where it was coming from.

[the remark this answers is not in the document]
Comment (reisr3) [reply frontend/src/components/smart-contracts/hooks/useERC20Events.ts:48]: And in this case, the gate doesn't need to be here. If `account` is required then we're good.

File: frontend/src/utils/schemas/v1.ts (hunk 1 above)
Comment (reisr3) [reply frontend/src/utils/schemas/v1.ts:15]: Yup :D

**Pull request merged**